crates/ruff_linter/src/rules/flake8_commas/rules/trailing_commas.rs

```diff
@@ -253,7 +253,11 @@ pub(crate) fn trailing_commas(
                     None
                 }
             }
-            _ => {
+            kind => {
+                if matches!(kind, TokenKind::Newline if fstrings > 0) {
+                    // The parser recovered from an unterminated f-string.
+                    fstrings = 0;
+                }
```

Comment:
I think this should work as the newline tokens within f-strings are actually `NonLogicalNewline`, I'll move this into `TokenIterWithContext`. I'll test this a lot because f-strings are complex.